### ISSUE TITLE
Terminating app due to uncaught exception

### ISSUE THREAD

**Issue by giulianok** [OneSignal/OneSignal-Cordova-SDK#723]
**Description:**
Upgraded to version 3.0.0 and the plugin stopped working on iOS and working as expected in Android.

Error is from
```
- (void)setLogLevel:(CDVInvokedUrlCommand*)command {
    [OneSignal setLogLevel:[command.arguments[0] intValue] visualLevel:[command.arguments[1] intValue]];
}
```

![Screen Shot 2021-08-23 at 8 33 28 PM](https://user-images.githubusercontent.com/8062245/130536952-7eb5eca1-2dd4-4286-9418-9800e1bb115c.png)

**Environment**

"@ionic-native/onesignal": "^5.35.0",

[info] Found 7 Capacitor plugins for android:
       @capacitor/browser@1.0.2
       @capacitor/camera@1.0.4
       @capacitor/device@1.0.2
       @capacitor/geolocation@1.0.2
       @capacitor/keyboard@1.0.2
       @capacitor/network@1.0.2
       @capacitor/status-bar@1.0.2
[info] Found 12 Cordova plugins for android:
       cordova-plugin-actionsheet@2.3.3
       cordova-plugin-androidx@3.0.0
       cordova-plugin-androidx-adapter@1.1.3
       cordova-plugin-app-version@0.1.12
       cordova-plugin-calendar@5.1.5
       cordova-plugin-firebase-analytics@5.0.0
       cordova-plugin-ionic@5.4.7
       cordova-plugin-market@1.2.0
       cordova-plugin-stripe@1.5.3
       cordova-plugin-whitelist@1.3.4
       cordova-support-android-plugin@1.0.2
       onesignal-cordova-plugin@3.0.0
[info] Found 7 Capacitor plugins for ios:
       @capacitor/browser@1.0.2
       @capacitor/camera@1.0.4
       @capacitor/device@1.0.2
       @capacitor/geolocation@1.0.2
       @capacitor/keyboard@1.0.2
       @capacitor/network@1.0.2
       @capacitor/status-bar@1.0.2
[info] Found 8 Cordova plugins for ios:
       cordova-plugin-actionsheet@2.3.3
       cordova-plugin-app-version@0.1.12
       cordova-plugin-calendar@5.1.5
       cordova-plugin-firebase-analytics@5.0.0
       cordova-plugin-ionic@5.4.7
       cordova-plugin-market@1.2.0
       cordova-plugin-stripe@1.5.3
       onesignal-cordova-plugin@3.0.0
[info] Found 4 incompatible Cordova plugins for ios, skipped install:
       cordova-plugin-androidx@3.0.0
       cordova-plugin-androidx-adapter@1.1.3
       cordova-plugin-whitelist@1.3.4
       cordova-support-android-plugin@1.0.2
[info] Listing plugins for web is not possible.

**Steps to Reproduce Issue:**

1. Install or Upgrade to 3.0.0
2. Use either Capacitor 2 or 3
3. Run iOS, either from xcode with simulator or real device as well from testflight

**Anything else:**

Logs

```
⚡️  [debug] - App render
⚡️  [debug] - App render content
⚡️  [debug] - App content
⚡️  [debug] - ready false
⚡️  [debug] - checkForUpdates
⚡️  [debug] - before maybeDeleteOldVersion
⚡️  [warn] - Cannot execute "reset" because "initialData" was not provided. Defaults to "remove"
⚡️  [debug] - App content
⚡️  [debug] - ready false
⚡️  [debug] - before newVersion
⚡️  [debug] - --WEBSOCKET undefined
⚡️  [debug] - App content
⚡️  [debug] - ready true
⚡️  To Native ->  Keyboard addListener 126966566
⚡️  To Native ->  Keyboard addListener 126966567
⚡️  To Native ->  Keyboard addListener 126966568
⚡️  To Native ->  Keyboard addListener 126966569
To Native Cordova ->  FirebaseAnalytics logEvent FirebaseAnalytics581699794 ["options": [page_view, {
    page = "/";
}]]
2021-08-23 20:33:16.873327-0400 App[813:642959] Starting Firebase Analytics plugin
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSDictionaryM intValue]: unrecognized selector sent to instance 0x283358720'
terminating with uncaught exception of type NSException
To Native Cordova ->  OneSignalPush setLogLevel OneSignalPush581699795 ["options": [{
    logLevel = 6;
    visualLevel = 0;
}, <null>]]
2021-08-23 20:33:16.884088-0400 App[813:642959] -[__NSDictionaryM intValue]: unrecognized selector sent to instance 0x283358720
2021-08-23 20:33:16.884672-0400 App[813:642959] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[__NSDictionaryM intValue]: unrecognized selector sent to instance 0x283358720'
*** First throw call stack:
(0x18af65654 0x18ac87bcc 0x18ae69dd8 0x18af697f8 0x18af6b71c 0x104bbaec8 0x105061f00 0x1050ab308 0x1050ab478 0x19266945c 0x19281c428 0x192a10010 0x1924cacfc 0x192754848 0x1924b795c 0x1924b7070 0x199e25bd8 0x199e25d74 0x18aee3ad8 0x18aee3a30 0x18aee31b8 0x18aede1e8 0x18aeddba8 0x19504d344 0x18f0193e4 0x104bb7cbc 0x18ad658f0)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by jkasten2** [comment]
@giulianok Thanks for reporting the issue. The parameters for `OneSignal.setLogLevel` have changed.

`setLogLevel` now accepts two params logLevel, visualLogLevel, instead of an object.
Example:
`window.plugins.OneSignal.setLogLevel({logLevel: 6, visualLevel: 0});` becomes `window.plugins.OneSignal.setLogLevel(6, 0);`.

https://documentation.onesignal.com/docs/cordova-30-api-reference#setloglevel

We will look into handling the old param format better for iOS since you noted this is not an issue for Android.

**Comment by giulianok** [comment]
@jkasten2 I see, so probably this is the error then https://github.com/ionic-team/ionic-native/blob/master/src/%40ionic-native/plugins/onesignal/index.ts#L705, I'm using Capacitor + Ionic

**Comment by giulianok** [comment]
Actually, might be an error in the OneSignal Cordova SDK, see https://github.com/OneSignal/OneSignal-Cordova-SDK/blob/main/src/ios/OneSignalPush.m#L270-L272, I think the Ionic Native package is just a wrapper

![Screen Shot 2021-08-24 at 2 48 30 PM](https://user-images.githubusercontent.com/8062245/130672899-2aaf5279-eb94-4321-b9dc-9dbe740b707f.png)

**Comment by giulianok** [comment]
I re-tested the lib, still having the issue, this is what I did:
1. Uninstalled the lib and ionic mapping: `npm uninstall onesignal-cordova-plugin @ionic-native/onesignal`
2. Removed all code that uses OneSignal
3. Built app and iOS Project: `npm run build && npx cap sync`
4. Ran it, everything working
5. Installed only OneSignal cordova plugin: `npm i --save onesignal-cordova-plugin@latest`
6. Built app and iOS Project: `npm run build && npx cap sync`
7. Error appears, app crashes
